brambling/utils/model_tables.py

```diff
@@ -374,11 +360,19 @@ def get_custom_fields(self):
     def _get_custom_fields(self):
         raise NotImplementedError
 
+    def _show_housing_data(self, attendee, form_entry):
+        if form_entry.form_field.form.form_type != 'housing':
+            return True
+        if attendee.needs_housing():
+            return True
+        else:
+            return False
+
     def get_field_val(self, obj, key):
         if key.startswith('custom_'):
             if not hasattr(obj, '_custom_data'):
                 raw_data = {
-                    entry.form_field_id: entry.get_value()
+                    entry.form_field_id: (entry.get_value() if self._show_housing_data(obj, entry) else '')
                     for entry in obj.custom_data.all()
```

Comment:
This should be `obj.custom_data.select_related('form_field__form'), otherwise it'll be 2nm additional queries (where n is the table size and m is the number of custom fields).